File: original/psqlConst.c
// source: https://github.com/postgres/postgres/blob/REL_13_STABLE/src/interfaces/libpq/libpq-fe.h#L112
typedef enum
{
	PQERRORS_TERSE,				/* single-line error messages */
	PQERRORS_DEFAULT,			/* recommended style */
	PQERRORS_VERBOSE,			/* all the facts, ma'am */
	PQERRORS_SQLSTATE			/* only error severity and SQLSTATE code */
} PGVerbosity;

// source: https://github.com/postgres/postgres/blob/REL_13_STABLE/src/interfaces/libpq/libpq-fe.h#L120
typedef enum
{
	PQSHOW_CONTEXT_NEVER,		/* never show CONTEXT field */
	PQSHOW_CONTEXT_ERRORS,		/* show CONTEXT for errors only (default) */
	PQSHOW_CONTEXT_ALWAYS		/* always show CONTEXT field */
} PGContextVisibility;

// source: https://github.com/postgres/postgres/blob/REL_13_STABLE/src/include/postgres_ext.h#L50
/*
 * Identifiers of error message fields.  Kept here to keep common
 * between frontend and backend, and also to export them to libpq
 * applications.
 */
#define PG_DIAG_SEVERITY		'S'
#define PG_DIAG_SEVERITY_NONLOCALIZED 'V'
#define PG_DIAG_SQLSTATE		'C'
#define PG_DIAG_MESSAGE_PRIMARY 'M'
#define PG_DIAG_MESSAGE_DETAIL	'D'
#define PG_DIAG_MESSAGE_HINT	'H'
#define PG_DIAG_STATEMENT_POSITION 'P'
#define PG_DIAG_INTERNAL_POSITION 'p'
#define PG_DIAG_INTERNAL_QUERY	'q'
#define PG_DIAG_CONTEXT			'W'
#define PG_DIAG_SCHEMA_NAME		's'
#define PG_DIAG_TABLE_NAME		't'
#define PG_DIAG_COLUMN_NAME		'c'
#define PG_DIAG_DATATYPE_NAME	'd'
#define PG_DIAG_CONSTRAINT_NAME 'n'
#define PG_DIAG_SOURCE_FILE		'F'
#define PG_DIAG_SOURCE_LINE		'L'
#define PG_DIAG_SOURCE_FUNCTION 'R'

File: src/translated/psqlConst.ts
// translated from ../../original/psqlConst.c
/* eslint-disable @typescript-eslint/naming-convention */
export enum PGContextVisibility {
  PQSHOW_CONTEXT_NEVER  = 0,     /* never show CONTEXT field */
  PQSHOW_CONTEXT_ERRORS = 1,     /* show CONTEXT for errors only (default) */
  PQSHOW_CONTEXT_ALWAYS = 2,     /* always show CONTEXT field */
}
export enum PGVerbosity {
  PQERRORS_TERSE    = 0,    /* single-line error messages */
  PQERRORS_DEFAULT  = 1,    /* recommended style */
  PQERRORS_VERBOSE  = 2,    /* all the facts, ma'am */
  PQERRORS_SQLSTATE = 3,    /* only error severity and SQLSTATE code */
}
export enum PGFieldCode {
  /* Identifiers of error message fields. */
  PG_DIAG_SEVERITY              = 'S',
  PG_DIAG_SEVERITY_NONLOCALIZED = 'V',
  PG_DIAG_SQLSTATE              = 'C',
  PG_DIAG_MESSAGE_PRIMARY       = 'M',
  PG_DIAG_MESSAGE_DETAIL        = 'D',
  PG_DIAG_MESSAGE_HINT          = 'H',
  PG_DIAG_STATEMENT_POSITION    = 'P',
  PG_DIAG_INTERNAL_POSITION     = 'p',
  PG_DIAG_INTERNAL_QUERY        = 'q',
  PG_DIAG_CONTEXT               = 'W',
  PG_DIAG_SCHEMA_NAME           = 's',
  PG_DIAG_TABLE_NAME            = 't',
  PG_DIAG_COLUMN_NAME           = 'c',
  PG_DIAG_DATATYPE_NAME         = 'd',
  PG_DIAG_CONSTRAINT_NAME       = 'n',
  PG_DIAG_SOURCE_FILE           = 'F',
  PG_DIAG_SOURCE_LINE           = 'L',
  PG_DIAG_SOURCE_FUNCTION       = 'R',
}
